transport_analysis/tests/analysis/test_velocityautocorr.py

```diff
@@ -1,63 +1,264 @@
 import pytest
-from numpy.testing import assert_allclose
+from numpy.testing import assert_almost_equal
 
-from transport_analysis.analysis.velocityautocorr import VelocityAutocorr
-from transport_analysis.tests.utils import make_Universe
+from transport_analysis.analysis.velocityautocorr import (
+    VelocityAutocorr as VACF,
+)
+import MDAnalysis as mda
+import numpy as np
 
+from MDAnalysis.exceptions import NoDataError
+from MDAnalysisTests.datafiles import PRM_NCBOX, TRJ_NCBOX
+from MDAnalysisTests.util import block_import, import_not_available
 
-class TestVelocityAutocorr:
 
+@pytest.fixture(scope="module")
+def u():
+    return mda.Universe(PRM_NCBOX, TRJ_NCBOX)
+
+
+@pytest.fixture(scope="module")
+def ag(u):
+    return u.select_atoms("name O and resname WAT and resid 1-10")
+
+
+@pytest.fixture(scope="module")
+def NSTEP():
+    nstep = 5000
+    return nstep
+
+
+@pytest.fixture(scope="module")
+def vacf(ag):
+    # non-fft VACF
+    v = VACF(ag, fft=False)
+    v.run()
+    return v
+
+
+# Step trajectory of unit velocities i.e. v = 0 at t = 0,
+# v = 1 at t = 1, v = 2 at t = 2, etc. for all components x, y, z
+@pytest.fixture(scope="module")
+def step_vtraj(NSTEP):
+    v = np.arange(NSTEP)
+    velocities = np.vstack([v, v, v]).T
+    # NSTEP frames x 1 atom x 3 dimensions, where NSTEP = 5000
+    velocities_reshape = velocities.reshape([NSTEP, 1, 3])
+    u = mda.Universe.empty(1, n_frames=NSTEP, velocities=True)
+    for i, ts in enumerate(u.trajectory):
+        u.atoms.velocities = velocities_reshape[i]
+    return u
+
+
+# Expected VACF results for step_vtraj
+# At time t, VACF is:
+# sum_{x=0}^{N - 1 - t} x*(x + t) * n_dim / n_frames
+# n_dim = 3 (typically) and n_frames = total_frames - t
+def characteristic_poly(last, n_dim, first=0, step=1):
+    diff = last - first
+    frames_used = diff // step + 1 if diff % step != 0 else diff / step
+    frames_used = int(frames_used)
+    result = np.zeros(frames_used)
+    for t in range(first, last, step):
+        sum = 0
+        sum = np.dtype("float64").type(sum)
+        lagtime = t - first
+        for x in range(first, (last - lagtime), step):
+            sum += x * (x + lagtime)
+        current_index = int(lagtime / step)
+        vacf = sum * n_dim / (frames_used - current_index)
+        result[current_index] = vacf
+    return result
+
+
+@block_import("tidynamics")
+def test_notidynamics(ag):
+    with pytest.raises(ImportError, match="tidynamics was not found"):
+        vacf = VACF(ag)
+        vacf.run()
+
+
+@pytest.mark.parametrize(
+    "tdim, tdim_keys", [(1, [0]), (2, [0, 1]), (3, [0, 1, 2])]
+)
+def test_characteristic_poly(step_vtraj, NSTEP, tdim, tdim_keys):
+    # test `characteristic_poly()` against `tidynamics.acf()``
+    try:
```

Comment:
Could you elaborate? Does that mean I should just remove the "try" and "except" part and only have the rest of the test?

Reply:
Yes as in you shouldn't be able to use the package without tidynamics tbh. I think we can be opinionated and insist. We didn't for MSD in MDAnalysis core because of the need to keep our dep list as short as possible

Reply:
So that would mean removing the exception and the corresponding test, along with the skip if no tidynamics parts in the tests, right? I've already added tidynamics in `devtools/conda-envs/test_env.yaml` and `pyproject.toml`. Would I add it to the `install_requires=` part of `setup.py` as well? (These are all from https://mdakits.mdanalysis.org/makingakit.html) Any other places?